service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsServiceTest.java

```diff
@@ -0,0 +1,235 @@
+
+import java.util.List;
+import java.util.Map;
+
+import static java.time.LocalDate.now;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.populatedCaseDetails;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    JacksonAutoConfiguration.class,
+    UploadDocumentsService.class,
+    FixedTimeConfiguration.class
+})
+class UploadDocumentsServiceTest {
+
+    private static final String USER = "HMCTS";
+    private CaseData givenCaseData;
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private UploadDocumentsService uploadDocumentsService;
+
+    @MockBean
+    private DocumentUploadHelper documentUploadHelper;
+
+    @BeforeEach
+    void setup() {
+        givenCaseData = prepareCaseData();
+        when(documentUploadHelper.getUploadedDocumentUserDetails()).thenReturn(USER);
+    }
+
+    @Test
+    void shouldSetUploadedByAndDateTimeOnCaseDocuments() {
+        CaseData caseData = caseData();
+
+        Map<String, Object> map = uploadDocumentsService.updateCaseDocuments(caseData, caseData);
+
+        assertThat(map.get("documents_socialWorkChronology_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .build());
+
+        assertThat(map.get("documents_socialWorkStatement_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .statusReason("Social work statement and genogram text")
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .documentStatus("To follow")
+                .build());
+
+        assertThat(map.get("documents_checklist_document"))
+            .isEqualToComparingOnlyGivenFields(Document.builder()
+                .statusReason("Social work chronology text")
+                .uploadedBy(USER)
+                .dateTimeUploaded(time.now())
+                .documentStatus("To follow")
+                .build());
+    }
+
+    @Test
+    void shouldUpdateOtherSocialWorkDocumentsListWithUpdatedDetailsAndUser() {
+        List<DocumentSocialWorkOther> list = unwrapElements(
+            uploadDocumentsService.setUpdatedByAndDateAndTimeOnDocuments(
+                createCaseDataWithUpdatedDocumentSocialWorkOther(),
+                createCaseDataWithOldDocumentSocialWorkOther()));
+
+        assertThat(list).first()
+            .extracting(DocumentSocialWorkOther::getTypeOfDocument)
+            .extracting(DocumentReference::getUrl)
```

Comment:
If you don't do it as a method reference you can probs combine these

Reply:
done